### PR TITLE
User's CRUD (soft deleting included)

### PR DESCRIPTION
**Pull request by ManuelOviedo**
- Note: migration refresh needed for already implemented instances
Resolves: User's CRUD

### CODE REVIEW

File: app/Http/Controllers/v1/UserController.php [hunk 1]
```diff
@@ -65,4 +67,138 @@ public function resetPassword(ResetPasswordRequest $request)
             'message' => trans($message)
         ], $httpCode);
     }
+  
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function list()
```
Comment (vik0x) [inline app/Http/Controllers/v1/UserController.php]:
```suggestion
    public function index()
```

File: app/Http/Controllers/v1/UserController.php [hunk 2]
```diff
@@ -65,4 +67,138 @@ public function resetPassword(ResetPasswordRequest $request)
...
+            "data"  => $collection
+        ], 200);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create(UserCreateRequest $request, User $user)
```
Comment (vik0x) [inline app/Http/Controllers/v1/UserController.php]:
```suggestion
    // if you want to update an user too, you could leave the $user var here, but you have a function to update the user below
    public function create(UserCreateRequest $request)
```

File: app/Http/Controllers/v1/UserController.php [hunk 3]
```diff
@@ -65,4 +67,138 @@ public function resetPassword(ResetPasswordRequest $request)
...
+    {
+        $user->fill($request->input())->save();
```
Comment (vik0x) [inline app/Http/Controllers/v1/UserController.php:105]: I preffer assign each attribute in the model to better understand of code

File: app/Http/Controllers/v1/UserController.php [hunk 4]
```diff
@@ -65,4 +67,138 @@ public function resetPassword(ResetPasswordRequest $request)
...
+     * @return \Illuminate\Http\Response
+     */
+    public function show(User $user)
+    {
+        return response()->json([
+            'data' => [
+                [
+                    "type"      => "user",
+                    "id"        => $user->id,
+                    "attributes"=> [
```
Comment (vik0x) [inline app/Http/Controllers/v1/UserController.php]: should we response with an array in the case where we need just one value?

File: app/Http/Controllers/v1/UserController.php [hunk 5]
```diff
@@ -65,4 +67,138 @@ public function resetPassword(ResetPasswordRequest $request)
...
+        ], 200);
+    }
+
+    /**
+     * Restore the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function restore($user)
```
Comment (vik0x) [inline app/Http/Controllers/v1/UserController.php:166]: Doesn't the dependencies injection work here?

File: app/Http/Requests/v1/UserCreateRequest.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    public function rules()
+    {
+        return [
+            'username'      => ['required', 'alpha_dash', 'unique:users,username'],
+            'email'         => ['required', 'email:rfc,dns', 'unique:users,email'],
+            'first_name'    => ['required', 'alpha'],
+            'middle_name'   => ['required', 'alpha'],
+            'last_name'     => ['required', 'alpha'],
+            'type'          => ['alpha'],
```
Comment (vik0x) [inline app/Http/Requests/v1/UserCreateRequest.php]: those fields are enum on the DB (type and status), so, we need to be more specific.
You could use a configuration value in config path of the project

File: app/Http/Requests/v1/UserEditRequest.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    public function rules()
+    {
+        return [
+            'username'      => ['alpha_dash'],
+            'email'         => ['email:rfc,dns'],
+            'first_name'    => ['alpha'],
+            'middle_name'   => ['alpha'],
+            'last_name'     => ['alpha'],
+            'type'          => ['alpha'],
```
Comment (vik0x) [inline app/Http/Requests/v1/UserEditRequest.php]: those fields are enum on the DB (type and status), so, we need to be more specific.
You could use a configuration value in config path of the project

**Review by vik0x — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/v1/UserController.php (hunk 5 above)
Comment (ManuelOviedo) [reply app/Http/Controllers/v1/UserController.php:166]: No. Eloquent ORM discriminates softDeleted records

**Review by vik0x — APPROVED** [review]

**Pull request merged**